infrastructure/running-a-node/setup-secure-wss.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Setup Secure WebSocket
+description: Instructions on enabling SSL for your node and setting up a secure WebSocket proxy server using nginx for remote connections.
+---
+
+# Setup Secure WebSocket
+
+## Introduction
+
+Ensuring secure WebSocket communication is crucial for maintaining the integrity and security of a Polkadot or Kusama node when interacting with remote clients. This guide walks you through setting up a secure WebSocket (WSS) connection for your node by leveraging SSL encryption with popular web server proxies like nginx or Apache. By the end of this guide, you'll be able to secure your node's WebSocket port, enabling safe remote connections without exposing your node to unnecessary risks.
+
+## Secure a WebSocket Port
+
+You can convert a non-secured WebSocket port to a secure WSS port by placing it behind an SSL-enabled proxy. This approach can be used to secure a bootnode or RPC server. The SSL-enabled apache2/nginx/other proxy server redirects requests to the internal WebSocket and converts it to a secure (WSS) connection. You can use a service like [LetsEncrypt](https://letsencrypt.org/){target=\_blank} to obtain an SSL certificate.
+
+### Obtain an SSL Certificate
+
+You can follow the LetsEncrypt instructions for your respective web server implementation to get a free SSL certificate:
+
+-  [nginx](https://certbot.eff.org/instructions?ws=nginx&os=ubuntufocal){target=\_blank}
+-  [apache2](https://certbot.eff.org/instructions?ws=apache&os=ubuntufocal){target=\_blank}
+ 
+LetsEncrypt will auto-generate an SSL certificate and include it in your configuration.
+
+You can generate a self-signed certificate and rely on your node's raw IP address when connecting. However, self-signed certificates aren't optimal because you have to whitelist the certificate to access it from a browser.
+
+Use the following commmand to generate a self-signed certificate using OpenSSL:
+
+--8<-- 'code/infrastructure/setup-secure-wss/install-openssl.md'
+
+## Install a Proxy Server
+
+There are a lot of different implementations of a WebSocket proxy; some of the more widely used are [nginx](https://www.nginx.com/){target=\_blank} and [apache2](https://httpd.apache.org/){target=\_blank}, both of which are commonly used web server implementations. See the following section for configuration examples for both implementations.
+
+### Use nginx
+
+1. Install the `nginx` web server: 
+    ```bash
+    apt install nginx
```

Comment:
Maybe we should add a note somewhere that all the steps presented in this guide are for Ubuntu-like oss

Reply:
Added a note for unix based systems